addOns/ascanrulesAlpha/src/test/java/org/zaproxy/zap/extension/ascanrulesAlpha/ForbiddenBypassScanRuleUnitTest.java

```diff
@@ -0,0 +1,217 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import static fi.iki.elonen.NanoHTTPD.newFixedLengthResponse;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import fi.iki.elonen.NanoHTTPD.IHTTPSession;
+import fi.iki.elonen.NanoHTTPD.Response;
+import org.junit.jupiter.api.Test;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.testutils.NanoServerHandler;
+
+/** Unit test for {@link ForbiddenBypassScanRule}. */
+public class ForbiddenBypassScanRuleUnitTest extends ActiveScannerTest<ForbiddenBypassScanRule> {
+
+    private static final String PROTECTED_PATH = "/protected/endpoint";
+
+    private static final String GENERIC_RESPONSE =
+            "<!DOCTYPE HTML PUBLIC \"-//IETF//DTD HTML 2.0//EN\">\n"
+                    + "<html><head></head><body></body></html>";
+
+    @Override
+    protected ForbiddenBypassScanRule createScanner() {
+        return new ForbiddenBypassScanRule();
+    }
+
+    @Test
+    public void shouldNotAlertIfInitialRequestIsNotForbidden() throws Exception {
+        // Given
+        String path = "/allowed/";
+        nano.addHandler(new OkResponse(path));
+        HttpMessage msg = this.getHttpMessage(path);
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(0));
+        assertEquals(1, httpMessagesSent.size());
+    }
+
+    @Test
+    public void shouldNotAlertIfAllRequestsAreForbidden() throws Exception {
+        // Given
+        String path = "/forbidden/";
+        nano.addHandler(new ForbiddenResponse(path));
+        HttpMessage msg = this.getHttpMessage(path);
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(0));
+        assertEquals(23, httpMessagesSent.size());
+    }
+
+    @Test
+    public void shouldAlertIfOkObtained() throws Exception {
+        // Given
+        nano.addHandler(new ForbiddenResponse(PROTECTED_PATH));
+        nano.addHandler(new OkResponse("/." + PROTECTED_PATH)); // Period is %2e
+        HttpMessage msg = this.getHttpMessage(PROTECTED_PATH);
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+        Alert alert = alertsRaised.get(0);
+        assertEquals("Bypassing 403", alert.getName());
+        assertEquals(Alert.RISK_MEDIUM, alert.getRisk());
+        assertEquals(Alert.CONFIDENCE_MEDIUM, alert.getConfidence());
+    }
+
+    @Test
+    public void shouldAlertIfOkWithHeaderPayload1() throws Exception {
```

Comment:
It's preferable to use specific names, e.g.`shouldAlertIfOkWithRewriteUrlHeader`.